hashes/zkevm/src/keccak/computation_circuit/circuit.rs

```diff
@@ -0,0 +1,343 @@
+        }
+        loaded_assigned_rows
+    }
+
+    // Generate compact inputs for Poseidon based on raw inputs in Keccak witnesses.
+    //
+    // To illustrate how this function works, let's take the following example:
+    // Parameters: RATE = 2, NUM_WORDS_TO_ABSORB = 2, NUM_ROUNDS = 3
+    // Considering the following logical input in little endian bytes:
+    // logical_input = [0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x21, 0x22, 0x23, 0x24, 0x25]
+    // Keccak witnesses represent the input in 2 chunks(NUM_ROUNDS + 1 rows per chunk):
+    // loaded_assigned_rows = [
+    //     {bytes_left:  21, word_value: 0x0807060504030201, is_final:   N/A},
+    //     {bytes_left:  13, word_value: 0x1817161514131211, is_final:   N/A},
+    //     {bytes_left: N/A, word_value:                N/A, is_final:   N/A},    // No input/output information after first NUM_WORDS_TO_ABSORB rows.
+    //     {bytes_left: N/A, word_value:                N/A, is_final: false},
+    //     {bytes_left:   5, word_value: 0x0000002524232221, is_final:   N/A},
+    //     {bytes_left:   0, word_value: 0x0000000000000000, is_final:   N/A},
+    //     {bytes_left: N/A, word_value:                N/A, is_final:   N/A},    // No input/output information after first NUM_WORDS_TO_ABSORB rows.
+    //     {bytes_left: N/A, word_value:                N/A, is_final:  true},
+    // ]
+    // For easier Poseidon processing, the raw input is encoded in the following way:
+    // If the logic input is empty, [0; RATE].
+    // If the logic input is empty, each word is encoded as RATE witnesses. First word is encoded
+    // as [<byte length of logical input>, logical_input[0], 0..]. Other words are encoded as [logical_input[i], 0, 0..].
+    // Note: With this encoding, App circuits should be able to compute Poseidons of variable length inputs easily.
+    // Then we get the following compact inputs for Poseidon hasher:
+    // poseidon_compact_inputs = [
+    //  {inputs: [                21, 0x0807060504030201], len: 2, is_final: false},   // 21 is the length of the logical input.
+    //  {inputs: [0x1817161514131211,                0x0], len: 2, is_final: false},
+    //  {inputs: [0x0000002524232221,                0x0], len: 2, is_final: true},    // The last row of the logical input.
+    //  {inputs: [               0x0,                0x0], len: 2, is_final: true},    // This row corresponds to the padding row in loaded_assigned_rows.
+    // ]
+    // The Poseidon results will be:
+    // poseidon_compact_outputs = [
+    //  {hash:                                N/A, is_final: false},
+    //  {hash:                                N/A, is_final: false},
+    //  {hash: <poseidon(<encoded logical input>), is_final:  true},
+    //  {hash:               poseidon([0x0, 0x0]), is_final:  true},
+    // ]
+    // Because of the padding rows, is_final cannot tell which row is the result of a logical input. Therefore
```

Comment:
Do you need this? `poseidon([0x0,0x0])` is the valid poseidon that matches `keccak([])`, so it's fine to include it?